### DISCUSSION TITLE
Integration of multiple-cucumber-html-reporter

### DISCUSSION THREAD

**Discussion by digital-h** [sorry-cypress/sorry-cypress#404] [Ideas]
Hi,

first of all: thank you very much for bringing this project to live. We are currently investigating if we can use it to als increase our cypress-tests and it looks very promising so far!

We are currently stumbling over one issue. As we use Cypress together with the Cucumber-Integration (https://github.com/TheBrainFamily/cypress-cucumber-preprocessor) we would also like to generate reports, which also relate to the Gherkin-Scenarios. Therefor we currently use the multiple-cucumber-html-reporter (https://github.com/wswebcreation/multiple-cucumber-html-reporter). 
Is there any chance to get this report also connected to sorry-cypress? We would also try to contribute if you show us the way how to do this best?
Basically what happens under the hood is, that the results are written during the execution of cucumber into different json-files. The cucumber-reporter then picks those json-files and creates a html-report out of that. Simple as that! But as you are working with GraphQL and Mongo in the background it is probably not only about passing files around.

Any ideas/insights are greatly welcome here!

**Comment by agoldis** [comment]
@digital-h 
Thanks for the kind words! 
To implement this feature we'd need to:

1. Find a way to upload the reports to remote storage. Today we get list of artifacts to upload from cypress agent, generate signed upload URLs and send back to cypress agent to run the actual upload. For uploading anything else - i.e. artifacts that aren't "native" to cypress, we'll need to implement an extra API for uploads + extra API for augmenting Runs results with extra data about "additional" artifacts that are not cypress-native.

2. Showing  / embedding the reports. That can be as simple as just shown a link to download or actually showing the report.


May be there's a simpler way of doing that, please let me know if you discover anything.

---

Looking at the reports, seems like the data sorry-cypress show is  60-70%  the same as cucumber report. some metadata is missing - e.g. OS/Device, and also the vocabulary is not "cucumber"-ish (e.g. tests vs features). Most of the data is actually present in the DB, but just not displayed. A possible contribution would be an "alternative" UI that shows the existing results as something that resembles cucumber report. I believe we can create very similar report based on data the we currently have without actually uploading the cucumber report
